File: test/code.py
filepath = './test/input.txt'
a = None

with open(filepath, encoding='utf8') as file:
    a = [line.strip() for line in file.readlines()]

res = 0

for s in a:
    a, b = None, None
    for i in range(len(s)):
        if s[i].isdigit():
            a = s[i]
            break
    
    for i in range(len(s)-1, -1, -1):
        if s[i].isdigit():
            b = s[i]
            break
    
    combine = a + b
    res += int(combine)

print(res)

   

File: test/code.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    string filepath = "../test/input.txt";
    vector<string> a;

    ifstream file(filepath);
    string line;

    while (getline(file, line)) {
        a.push_back(line);
    }

    file.close();

    int res = 0;

    for (string& s : a) {
        char x, y;

        for (int i = 0; i < s.size(); i++) {
            if (isdigit(s[i])) {
                x = s[i];
                break;
            }
        }

        for (int i = s.size()-1; i >= 0; i--) {
            if (isdigit(s[i])) {
                y = s[i];
                break;
            }
        }

        string temp = "";
        temp += x;
        temp += y;

        res += stoi(temp);
    }

    cout << res << "\n";
    return 0;
}